packages/go_router/lib/src/router.dart

```diff
@@ -496,12 +496,17 @@ class GoRouter implements RouterConfig<RouteMatchList> {
   ///
   /// If the top-most route is a pop up or dialog, this method pops it instead
   /// of any GoRoute under it.
+  ///
+  /// @restore()
+  /// Ensure that the `value` of `routeInformationProvider` is synced
+  ///  with `routerDelegate.currentConfiguration`.
   void pop<T extends Object?>([T? result]) {
     assert(() {
       log('popping ${routerDelegate.currentConfiguration.uri}');
       return true;
     }());
     routerDelegate.pop<T>(result);
+    restore(routerDelegate.currentConfiguration);
```

Comment:
can you write a test for this, I know this is covered in example test, but they may be updated without knowing it no longer validate the fix here

Reply:
done